src/containers/edit-playlist/mobile/EditPlaylistPage.tsx

```diff
@@ -179,15 +199,30 @@ const EditPlaylistPage = g(
       if (formFields.description === undefined) {
         formFields.description = null
       }
+      // Copty the metdata playlist contents so that a reference is not changed between
+      // removing tracks, updating orderm and edit playlist
+      const playlistTrackIds = [
+        ...(metadata?.playlist_contents?.track_ids ?? [])
+      ]
+
+      for (const removedTrack of removedTracks) {
+        const { playlist_id } = metadata!
+        removeTrack(removedTrack.trackId, playlist_id, removedTrack.timestamp)
+      }
 
       if (metadata && formFields.playlist_id) {
         // Edit playlist
         if (hasReordered) {
-          // Reorder the playlist and refresh the linup just in case it's
+          // Reorder the playlist and refresh the lineup just in case it's
           // in the view behind the edit playlist page.
           orderPlaylist(
             metadata.playlist_id,
-            formatReorder(metadata, reorderedTracks)
+            formatReorder(playlistTrackIds, reorderedTracks)
+          )
+          // Update the playlist content track_ids so that the editPlaylist
+          // optimistically update the cached collection trackIds
+          formFields.playlist_contents.track_ids = reorderedTracks.map(
```

Comment:
I needed to do this so that line 229 `editPlaylist` would contain the correct reordered tracks, else it would incorrectly display the track order until the playlist changes are confirmed.